app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -1,54 +1,65 @@
 module EmsRefresh::SaveInventoryHelper
-  def save_inventory_multi(type, parent, hashes, deletes, find_key, child_keys = [], extra_keys = [])
+  # first argument is either an association or a symbol and the parent class
+  # save_inventory_multi(hardware.guest_devices, hashes, deletes, [:uid_ems], [:miq_scsi_targets], [:switch, :lan])
+  # save_inventory_multi(:guest_devices, hardware, hashes, deletes, [:uid_ems], [:miq_scsi_targets], [:switch, :lan])
+  def save_inventory_multi(type, parent, hashes, deletes, find_key = [], child_keys = [], extra_keys = [])
+    association = nil
+    if !type.kind_of?(Symbol) && type < ActiveRecord::Associations
+      association = type
+      # shift args left one
+      raise IllegalArgument, "too many parameters" unless extra_keys.blank?
+      hashes, deletes, find_key, child_keys, extra_keys = parent, hashes, deletes, find_key, child_keys
+    else
+      # legacy accessor
```

Comment:
I think I'd rather just change this, and deal with the obvious breakage if there are any missed callers.

But if we're going to keep this compatibility, the legacy spelling should be deprecated, shouldn't it?